### ISSUE TITLE
[SupplyParameterFromQuery] property not being set during OnInitializedAsync

### ISSUE THREAD

**Issue by MohamedTwheed** [dotnet/aspnetcore#53055]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

After migrating from .net7 to .net8 I encountered an issue that properties with [SupplyParameterFromQuery] not being set druing the OnInitializedAsync only after OnParametersSetAsync.

Ex of the bug:
Navigation with: https://localhost:44340/test?CrumbTrailItemId=662f20c8-b7c1-4145-a8cd-ad693913414a
```
    [SupplyParameterFromQuery]
    public string CrumbTrailItemId { get; set; }

    protected async Task OnInitializedAsync()
    {
        // this.CrumbTrailItemId  is null
    }

    protected async Task OnParametersSetAsync()
    {
        // this.CrumbTrailItemId  is "662f20c8-b7c1-4145-a8cd-ad693913414a"
    }
```

### Expected Behavior

The property with annotation SupplyParameterFromQuery should be set at the beginning of the life cycle of the component but somehow its being set after OnParametersSetAsync.

Ex of expected behavior:
Navigation with: https://localhost:44340/test?CrumbTrailItemId=662f20c8-b7c1-4145-a8cd-ad693913414a
```
    [SupplyParameterFromQuery]
    public string CrumbTrailItemId { get; set; }

    protected async Task OnInitializedAsync()
    {
        // this.CrumbTrailItemId  is "662f20c8-b7c1-4145-a8cd-ad693913414a"
    }
```

### .NET Version

8.0.0

### Anything else?

none

**Comment by symbiotic** [comment]
i'm using noticing similar issues after moving from .NET 7 to .NET 8. with the parameter not always set, but also OnInitializedAsync not always firing. i'm using the component as a page though (with the page directive @page). i'll have to find time to create a separate issue, because this is a huge problem for me

**Comment by hakenr** [comment]
related #51398

**Comment by MackinnonBuck** [comment]
Tried this myself and in my basic test app, I was able to access updated values of `[SupplyParameterFromQuery]` properties in `OnInitializedAsync()`. This was true both when refreshing the page and navigating from another page.

Could you provide a repro project so we can investigate the issue further?

**Comment by rabberbock** [comment]
@MackinnonBuck Here is a repro. https://github.com/rabberbock/SupplyQueryBugNet8/tree/main/BlazorApp

The issue surfaces when the page you are navigated from has `SupplyParameterFromQuery` and the page you are going to has one as well. 

https://github.com/rabberbock/SupplyQueryBugNet8/blob/main/BlazorApp/Pages/Counter.razor#L16
https://github.com/rabberbock/SupplyQueryBugNet8/blob/main/BlazorApp/Pages/Home.razor#L22

https://github.com/dotnet/aspnetcore/assets/3247622/9708a7da-828e-42bc-a2e9-06ee0441452c

**Comment by kyletinsley** [comment]
I can confirm this issue is still occurring in 8.0.1.

**Comment by milkyjoe90** [comment]
I'm also experiencing this issue in 8.0 and 8.0.1, have reverted back to 7 in order to continue developing.

**Comment by MohamedTwheed** [comment]
This issue still exist and as a workaround i've to manually parse out the parameters inside OnInitialized for the first run.
any stable solution?

**Comment by rabberbock** [comment]
@MackinnonBuck Can we please get this issue reopened? I put up a repro above.  Thanks!

**Comment by kyletinsley** [comment]
Why is this closed? This change breaks our app and we cannot upgrade to 8.0.1.

**Comment by mkArtakMSFT** [comment]
Thanks for raising this issue, everyone. Looks like we had left a gap in the original PR. We're going to address it in the next servicing release (probably 8.0.3). I will update the milestone to the target release, as soon as we get the servicing PR merged.

**Comment by mikecasas** [comment]
Any update on this being assigned a milestone for release.

**Comment by hakenr** [comment]
The PR #53625 is associated to 8.0.3 milestone.
